scanner/src/main/kotlin/scanners/fossid/Utils.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners.fossid
+
+import org.ossreviewtoolkit.clients.fossid.model.rules.IgnoreRule
+import org.ossreviewtoolkit.clients.fossid.model.rules.RuleType
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.PathExclude
+
+private val DIRECTORY_PATTERN = "((?:\\w+/)*\\w+)/?(\\*\\*)?".toRegex()
+private val EXTENSION_PATTERN = "\\*\\.(\\w+)".toRegex()
+private val FILE_PATTERN = "([\\w.]+)".toRegex()
```

Comment:
Consider using named capturing groups for readability.

Reply:
Also: These are not patterns / strings, but compiled regexes, so the variable names should end in `_REGEX`.